modules/core/02-client/keeper/extended_keeper.go

```diff
@@ -27,6 +24,28 @@ type ExtendedKeeper struct {
 	hooks ClientHooks
 }
 
+// GetHeaderInfo returns the information necessary for header timestamping or nil
+// if provided message is not a header
+func (ek ExtendedKeeper) GetHeaderInfo(ctx sdk.Context, m exported.ClientMessage) *HeaderInfo {
```

Comment:
```suggestion
func GetHeaderInfo(ctx sdk.Context, m exported.ClientMessage) *HeaderInfo {
```

It seems that this function does not need to be attached to the keeper